Ref #543 -- Allow send mail from order

danlls opened the pull request:
- Add send email directly for order
- Add email history (from mass and custom) to each specific order

** Need help to identify issue on testing, mail doesn't seems to be sent
  in the test cases. However, testing it using a browser with a running
  python debugging SMTP server works perfectly fine as the email is printed
  on the shell stdout and alert-success appear on redirected page.

raphaelm commented on src/pretix/control/forms/orders.py:
```diff
@@ -178,3 +179,31 @@ def __init__(self, *args, **kwargs):
+            widget=I18nTextInput, required=True,
+            initial=order.email, locales=event.settings.get('locales')
+        )
+        self.fields['sendto'].widget.attrs['readonly'] = 'readonly'
```
We don't really need I18nFormFields here, since when we write to one attendee only, we will only use one language. You can just use plain CharFields/TextFields/EmailFields and then you can define them like any other django form field

danlls commented:
@raphaelm Thanks! it make sense to use plain fields because the email only needs to be localized to the user but not the author. Everything is resolved now using plain django form fields.

raphaelm commented on src/pretix/control/logdisplay.py:
```diff
@@ -96,6 +96,7 @@ def pretixcontrol_logentry_display(sender: Event, logentry: LogEntry, **kwargs):
         'pretix.event.order.payment.changed': _('The payment method has been changed.'),
         'pretix.event.order.expire_warning_sent': _('An email has been sent with a warning that the order is about '
                                                     'to expire.'),
+        'pretix.event.order.mail_sent': _('The order received a custom email.'),
```
Better: A custom email has been sent. An order is not something that can receive emails ;)

danlls commented:
@raphaelm Thanks for the feedback, fixed point 1 and 3. I agree with point 4 to include other mails in history but in another PR.

Other modification:
- Renamed placeholder and email_context keys to be similar with other existing placeholders for consistency.

danlls commented:
@raphaelm latest master branch is failing pytest in my local machine.

rixx commented:
It is fixed now :)

raphaelm commented on src/pretix/control/templates/pretixcontrol/order/mail_history.html:
```diff
@@ -0,0 +1,33 @@
+    <h1>{% trans "Email history" %}</h1>
+    <div class="callout callout-info">
+        <p>Includes sent custom and mass emails(<em>plugin</em>) history only.</p>
+    </div>
```
Please use ``alert alert-info`` instead of defining a new UI pattern and use ``{% trans "…" %}`` for the text.

(Also, I'd drop the "(plugin)", I don't think its helpful)

raphaelm commented on src/pretix/control/forms/orders.py:
```diff
@@ -215,3 +215,30 @@ def __init__(self, *args, **kwargs):
+            label=_("Message"),
+            required=True,
+            widget=forms.Textarea,
+            initial=order.event.settings.mail_text_order_custom_mail,
```
we should probably use ``order.event.settings.mail_text_order_custom_mail.localize(order.locale)`` here

danlls commented:
As you've mention we should log the message in one language in `SenderView`, which make logical sense as each order only receives the message in their specified locale. Then, event would log the message in multiple locales. 
 
The logs you're looking at are the old logs. Try sending new mass mail it should work correctly.

raphaelm commented:
> The logs you're looking at are the old logs. Try sending new mass mail it should work correctly.

:see_no_evil: You caught me. 

Of course, real users also have old logs out there, but I'll find a way to deal with that.

Thank you very much for this high-quality pull request! :heart:
